### PR TITLE
backendutil: Update to go-message 0.10

### PR DESCRIPTION
**Pull request by foxcpp**

### CODE REVIEW

**Comment by emersion** [comment]
Eh, this looks like a go-message bug…

**Comment by emersion** [comment]
Can you confirm https://github.com/emersion/go-message/commit/acbd4f2ec4a63ad762e19df18637c6c096eb8395 fixes it?

**Comment by foxcpp** [comment]
It doesn't seem to fix it.
```
~/Projects/go-imap · (go-message-0.10±)
⟩ cat go.mod
module github.com/emersion/go-imap

require (
	github.com/emersion/go-message v0.10.2-0.20190514190358-acbd4f2ec4a6
	github.com/emersion/go-sasl v0.0.0-20161116183048-7e096a0a6197
	golang.org/x/text v0.3.2
)

~/Projects/go-imap · (go-message-0.10±)
⟩ go test -v github.com/emersion/go-imap/backend/backendutil -count 1
=== RUN   TestFetchBodySection
=== RUN   TestFetchBodySection/BODY[]
=== RUN   TestFetchBodySection/BODY[1.1]
=== RUN   TestFetchBodySection/BODY[1.2]
=== RUN   TestFetchBodySection/BODY[2]
=== RUN   TestFetchBodySection/BODY[HEADER]
=== RUN   TestFetchBodySection/BODY[HEADER.FIELDS_(From_To)]
panic: message: field not found in Header.m [recovered]
	panic: message: field not found in Header.m

goroutine 12 [running]:
testing.tRunner.func1(0xc0000ba700)
	/usr/lib/go/src/testing/testing.go:830 +0x392
panic(0x5719c0, 0x5df260)
	/usr/lib/go/src/runtime/panic.go:522 +0x1b5
github.com/emersion/go-message/textproto.(*headerFields).Del(0xc00004cda0)
	/home/foxcpp/.go/pkg/mod/github.com/emersion/go-message@v0.10.2-0.20190514190358-acbd4f2ec4a6/textproto/header.go:180 +0x547
github.com/emersion/go-imap/backend/backendutil.FetchBodySection(0xc00009e550, 0xc000091f20, 0x0, 0x0, 0x0, 0x0)
	/home/foxcpp/Projects/go-imap/backend/backendutil/body.go:60 +0x8e6
github.com/emersion/go-imap/backend/backendutil.TestFetchBodySection.func1(0xc0000ba700)
	/home/foxcpp/Projects/go-imap/backend/backendutil/body_test.go:108 +0x223
testing.tRunner(0xc0000ba700, 0xc0000139b0)
	/usr/lib/go/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/lib/go/src/testing/testing.go:916 +0x35a
FAIL	github.com/emersion/go-imap/backend/backendutil	0.005s

```

**Comment by emersion** [comment]
Gah, another bug.

File: backend/backendutil/body.go [hunk 1]
```diff
@@ -43,10 +43,7 @@ func FetchBodySection(e *message.Entity, section *imap.BodySectionName) (imap.Li
 	header := e.Header
 	if section.Fields != nil {
 		// Copy header so we will not change message.Entity passed to us.
-		header = make(message.Header, len(e.Header))
-		for k, v := range e.Header {
-			header[k] = v
-		}
+		header := e.Header.Copy()
```
Comment (emersion) [inline backend/backendutil/body.go]: This needs to be `=`, not `:=`

**Comment by emersion** [comment]
Pushed a fix for the second bug: https://github.com/emersion/go-message/commit/39c160d62e796e915ea3fd17534320035c0bcf9c

Sorry about that.

**Comment by foxcpp** [comment]
Oh, well, now it works but fields order is reversed. :\

**Comment by emersion** [comment]
Continuing the "go-message is broken" series:

go-message still used mime/multipart for reading/writing multipart messages. Switched to an internal reader/writer here:

https://github.com/emersion/go-message/commit/604a7372299371ce25c2a4dff3cbb2675a65a514

**Comment by foxcpp** [comment]
```
cmd_selected_test.go:225: Invalid FETCH response: * 1 FETCH (BODY[TEXT] {171}
```

Looks like `FETCH 1 (BODY.PEEK[TEXT])` response includes header too.

After some debugging, it seems like `e.Body` includes the message header.
```
	bd, err := ioutil.ReadAll(e.Body)
	log.Println("BODY", string(bd), err)
```
before `// Write the body, if requested` in backendutil/body.go (FetchBodySection) prints it.

**Comment by emersion** [comment]
This seems to be related to non-CRLF line endings used in the test messages.

**Comment by foxcpp** [comment]
So I should update them.

**Comment by emersion** [comment]
Yeah, this would be better. It's in `backend/memory/backend.go`. I'll add a test to go-message to make sure we handle this better.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**